From de50be6dde3be057e213679fdc33c0b05090c609 Mon Sep 17 00:00:00 2001
From: Liang Yan <lyan@suse.com>
Date: Fri, 26 Jan 2018 16:22:46 -0500
Subject: [PATCH] Kabi: Keep KVM stable after enable s390 wire up bpb feature
Patch-mainline: no, kabi patches are downstream
References: bsc#1076805

The patch "patches.arch/KVM-s390-wire-up-bpb-feature.patch"
changed  two structs, however both structs are only extended,
either a previously reserved fields is now being used
or a new field is added at the end. The overall size of the
structure does not change (the sie_block stays at 512
and the sync_regs is only used as part of the kvm_run structure
which also stays the same. So include __GENKSYMS__ to hide it from
the kABI checker.

Signed-off-by: Liang Yan <lyan@suse.com>
---
 arch/s390/include/asm/kvm_host.h | 6 +++++-
 arch/s390/include/uapi/asm/kvm.h | 2 ++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/s390/include/asm/kvm_host.h b/arch/s390/include/asm/kvm_host.h
index 76737b6adbbd..64a7c12e16a1 100644
--- a/arch/s390/include/asm/kvm_host.h
+++ b/arch/s390/include/asm/kvm_host.h
@@ -136,8 +136,12 @@ struct kvm_s390_sie_block {
 	__u16	ipa;			/* 0x0056 */
 	__u32	ipb;			/* 0x0058 */
 	__u32	scaoh;			/* 0x005c */
-#define FPF_BPBC 	0x20
+#ifdef __GENKSYMS__
+        __u8	reserved60;		/* 0x0060 */
+#else
+#define FPF_BPBC	0x20
 	__u8	fpf;			/* 0x0060 */
+#endif
 	__u8	ecb;			/* 0x0061 */
 	__u8    ecb2;                   /* 0x0062 */
 #define ECB3_AES 0x04
diff --git a/arch/s390/include/uapi/asm/kvm.h b/arch/s390/include/uapi/asm/kvm.h
index 60f92c0761ce..cff397553996 100644
--- a/arch/s390/include/uapi/asm/kvm.h
+++ b/arch/s390/include/uapi/asm/kvm.h
@@ -169,7 +169,9 @@ struct kvm_sync_regs {
 	__u64 vrs[32][2];	/* vector registers */
 	__u8  reserved[512];	/* for future vector expansion */
 	__u32 fpc;	/* only valid with vector registers */
+#ifndef __GENKSYMS__
 	__u8 bpbc : 1;		/* bp mode */
+#endif
 };
 
 #define KVM_REG_S390_TODPR	(KVM_REG_S390 | KVM_REG_SIZE_U32 | 0x1)
-- 
2.16.1

